### ISSUE TITLE
Issue Getting the code to utilize google collab GPU

### ISSUE THREAD

**Issue by ParsaMokhtariHessar** [makgyver/gossipy#2]
Hi, I'm Parsa a master's student at the university of Semnan.
I am trying to get gossipy (particularly [main_onoszko_2021.py](https://github.com/makgyver/gossipy/blob/main/main_onoszko_2021.py)) to run In google collab. although successful; it takes long training times and shows 0 utilization of GPU . that is when CUDA is available for PyTorch. and also GPU is enabled in collab. let me know if you wanna take a look at my code I will email it to you.
btw I have tried solutions like (numba @jit(target_backend='cuda')  ). if you can put me in the right path. may tell me how you managed to get it to use GPU? I will be grateful

**Comment by makgyver** [comment]
Hi Parsa, thank you for your interest in gossipy.
Currently, gossipy does not support GPUs; however, I am planning to add it soon. 
It should be rather a quick fix; unfortunately, I did not have much time to work on it.

**Comment by makgyver** [comment]
This issue has been fixed in [8bd4fbf](https://github.com/makgyver/gossipy/commit/8bd4fbff9982ca88d1d7abec960f5dab8989c013)

It is possible to set the device using the GlobalSetting singleton object:
```python
from gossipy import GlobalSettings
GlobalSettings().set_device("cuda")
```
**before** generating the clients and setting up the simulation. I suggest putting it at the beginning of your python scripts.

This will be included in the next release. For now, is available only by cloning the repo.
